src/NetworkService.cpp

```diff
@@ -129,27 +126,23 @@ void NetworkService::handle_dataset_lock_req(WorkerContext *wctx,
             spdlog::info("Coordinator: dataset {} locked", ds_name);
         }
 
-        s_send_val<NetLockResp>(backend, NetLockResp::LockOk);
+        s_send<NetLockResp>(&backend, NetLockResp::LockOk);
     } else {
         spdlog::warn("Coordinator: dataset lock denied");
-        s_send_val<NetLockResp>(backend, NetLockResp::LockDenied);
+        s_send<NetLockResp>(&backend, NetLockResp::LockDenied);
     }
 }
 
 void NetworkService::handle_iterator_lock_req(WorkerContext *wctx,
                                               const std::string &worker_addr) {
-    if (s_recv(backend).size() != 0) {
-        throw std::runtime_error("Expected zero-size frame");
-    }
+    s_recv_padding(&backend);
 
-    std::string iterator_name = s_recv(backend);
+    auto iterator_name = s_recv<std::string>(&backend);
```

Comment:
In a lot of places in this PR: maybe consider?
```suggestion
    auto iterator_name { s_recv<std::string>(&backend) };
```